### PR TITLE
Resort to using `isNative` check over `isStructured`

### PR DESCRIPTION
**Pull request by nemanjaglumac**
This PR addresses semantics only. The logic of the code it touches should remain the same as before this change.

A follow up PR that should address this non-blocking comment
https://github.com/metabase/metabase/pull/37977#discussion_r1467360379

### CODE REVIEW

File: frontend/src/metabase-lib/Question.ts [hunk 1]
```diff
@@ -557,9 +557,9 @@ class Question {
   }
 
   usesSegment(segmentId): boolean {
-    const isStructured = !Lib.queryDisplayInfo(this.query()).isNative;
+    const { isNative } = Lib.queryDisplayInfo(this.query());
```
[the remark this answers is not in the document]
Comment (oleggromov) [reply frontend/src/metabase-lib/Question.ts:560]: Probably it will convey the intentions better because we will have less negations.

**Review by oleggromov — APPROVED** [review]
LGTM 👍
Added a small suggestion but it's not a blocker.

File: frontend/src/metabase-lib/Question.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (nemanjaglumac) [reply frontend/src/metabase-lib/Question.ts:560]: So you basically reverted the diff?
This is where I started from.

The internal deal was made to NOT use `isStructured` and to go with `isNative` everywhere.
I don't like negations in conditionals either, but... 🤷‍♂️

[the remark this answers is not in the document]
Comment (oleggromov) [reply frontend/src/metabase-lib/Question.ts:560]: Sorry for the confusion, all good!

**Pull request merged**